xwiki-rendering-api/src/main/java/org/xwiki/rendering/parser/ParseException.java

```diff
@@ -41,6 +43,20 @@ public ParseException(String message)
         super(message);
     }
 
+    /**
+     * Constructs a new parse exception with the specified cause .
+     *
+     * @param cause the cause (which is saved for later retrieval by the {@link #getCause()} method).)
+     * @since 14.9RC1
+     * @since 14.4.6
+     * @since 13.10.10
+     */
+    @Unstable
+    public ParseException(Throwable cause)
```

Comment:
I find this constructor very rarely needed, as I feel it's a good practice to always associated a message with the exception.

Reply:
That the first time I have a use for this, because if failed to find a message to add to the `StackOverflowError` catch block. 
Unless we lean for something a bit redundant and generic like "Failed to parse input source"?

Reply:
Anyway, as I said in another comment the protection already exist somewhere, it just needs to include StackOverflowError so you don't really need that constructor.

Reply:
Yep, just realized that :) Fixed